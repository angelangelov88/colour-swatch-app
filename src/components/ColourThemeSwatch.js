import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { CheckCircleIcon, BanIcon } from '@heroicons/react/outline'
import { useState } from 'react'


const ColourThemeSwatch = (props) => {
  let {
    color,
    className
  } = props
  const [clicked, setClicked] = useState(false)

  const clickFunction = () => {
    console.log(`${color}`)

    setClicked(!clicked)
    console.log(clicked)
  }

  const classes = classNames(
    clicked && "text-white",

    "font-ibm flex items-center justify-center",

    !clicked && "hover:outline hover:outline-4",
    !clicked && "hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)]",
    // "hover:shadow-[0_1px_2px_rgba(112,53,13,0.06)]",
    
    "group hover:cursor-pointer",
    
    !clicked && "active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)]",

    "w-6 h-6 m-2 rounded border border-solid",

    color === 'primary' && 'bg-primary-light text-primary-dark active:text-primary-90 border-primary-dark outline-primary-30',
    
    // color === 'gray' && 'bg-gray-60 text-gray-80 active:text-gray-90 border-gray-80 outline-gray-30',
    // color === 'purple' && 'bg-purple-60 text-purple-80 active:text-purple-90 border-purple-80 outline-purple-30',
    // color === 'blue' && 'bg-blue-60 text-blue-80 active:text-blue-90 border-blue-80 outline-blue-30',
    // color === 'lightblue' && 'bg-lightblue-60 text-lightblue-80 active:text-lightblue-90 border-lightblue-80 outline-lightblue-30',
    // color === 'cyan' && 'bg-cyan-60 text-cyan-80 active:text-cyan-90 border-cyan-80',
    // color === 'teal' && 'bg-teal-60 text-teal-80 active:text-teal-90 border-teal-80',
    // color === 'green' && 'bg-green-60 text-green-80  active:text-green-90 border-green-80 outline-green-30',
    // color === 'yellow' && 'bg-yellow-60 text-yellow-80 active:text-yellow-90 border-yellow-80 outline-yellow-30',
    // color === 'red' && 'bg-red-60 text-red-80 active:text-red-90 border-red-80 outline-red-30',
    
    color && `bg-${color}-60 text-${color}-80 active:text-${color}-90 border-${color}-80 outline-${color}-30`,
    
    
    color === 'void' && 'bg-white text-gray-60 active:text-primary-90 border-gray-80 outline-primary-30',

    

    className
    // variant === "primary" && "disabled:text-primary-10 text-white hover:bg-primary-blue active:bg-primary-dark disabled:bg-primary-30 bg-primary-light shadow disabled:border-primary-20 border-primary-dark",
  );


  return (
    <div>
      <p>{color}</p>
      <div
        className={classes} 
        onClick={clickFunction}
      >
      <span className={classNames(
        "group-hover:block",
        clicked ? "block text-white" : "hidden"
      )}>
      {color === 'void' ?
        <BanIcon className={classNames(
          "text-gray-60"
        )}/>

         : <CheckCircleIcon className={classNames(`w-5 h-5`)} />
        }
        
        
      </span>
      </div>
      <div>

      
        </div>

    </div>
  )
}

ColourThemeSwatch.defaultProps = {
  color: "primary",
}

ColourThemeSwatch.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    "gray",
    "primary",
    "purple",
    "blue",
    "lightblue",
    "cyan",
    "teal",
    "green",
    "yellow",
    "red",
    "void",
  ]),
  
}

export { ColourThemeSwatch }
