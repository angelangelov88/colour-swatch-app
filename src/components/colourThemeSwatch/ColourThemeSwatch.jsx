import React from "react"
import classNames from "classnames"
import PropTypes from "prop-types"
import { CheckCircleIcon, BanIcon } from "@heroicons/react/outline"

const ColourThemeSwatch = (props) => {
  const { color, className } = props
  const [ clicked, setClicked ] = React.useState(false)
  const handleClick = () => setClicked(!clicked)

  const classes = classNames(
    "flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer",
    !clicked && "hover:outline hover:outline-4",
    !clicked && "active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)]",
    !clicked && "hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)]",
    color === "gray" && "bg-gray-60 text-gray-80 active:text-gray-90 border-gray-80 outline-gray-30",
    color === "primary" && "bg-primary-60 text-primary-80 active:text-primary-90 border-primary-80 outline-primary-30",
    color === "purple" && "bg-purple-60 text-purple-80 active:text-purple-90 border-purple-80 outline-purple-30",
    color === "blue" && "bg-blue-60 text-blue-80 active:text-blue-90 border-blue-80 outline-blue-30",
    color === "lightblue" && "bg-lightblue-60 text-lightblue-80 active:text-lightblue-90 border-lightblue-80 outline-lightblue-30",
    color === "cyan" && "bg-cyan-60 text-cyan-80 active:text-cyan-90 border-cyan-80 outline-cyan-30",
    color === "teal" && "bg-teal-60 text-teal-80 active:text-teal-90 border-teal-80 outline-teal-30",
    color === "green" && "bg-green-60 text-green-80 active:text-green-90 border-green-80 outline-green-30",
    color === "yellow" && "bg-yellow-60 text-yellow-80 active:text-yellow-90 border-yellow-80 outline-yellow-30",
    color === "red" && "bg-red-60 text-red-80 active:text-red-90 border-red-80 outline-red-30",
    color === "void" && "bg-white text-gray-60  border-gray-60 outline-primary-30 active:border-gray-80 active:text-gray-90",
    color === "void" && !clicked && "hover:text-gray-70",
    className
  );

  return (
    <div
    className={classes} 
    onClick={handleClick}
    >
    {color === "void" 
      ?
      <span className="w-5 h5">
        {color === 'void' && !clicked ? <BanIcon /> : <CheckCircleIcon />}
      </span>
      :
      <span className={classNames(
        "group-hover:block",
        clicked ? "block text-white" : "hidden"
      )}>
        <CheckCircleIcon className="w-5 h-5" />
      </span>
    }
    </div>
  )
}

ColourThemeSwatch.defaultProps = {
  color: "primary",
}

ColourThemeSwatch.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    "gray",
    "primary",
    "purple",
    "blue",
    "lightblue",
    "cyan",
    "teal",
    "green",
    "yellow",
    "red",
    "void"
  ]),

}

export { ColourThemeSwatch }