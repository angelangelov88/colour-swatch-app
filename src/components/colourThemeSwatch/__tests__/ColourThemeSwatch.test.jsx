import React from 'react';
import renderer from 'react-test-renderer';
import { ColourThemeSwatch } from '../ColourThemeSwatch';

it('should render with color gray', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="gray" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color primary', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="primary" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color purple', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="purple" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color blue', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="blue" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color lightblue', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="lightblue" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color cyan', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="cyan" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color teal', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="teal" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color green', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="green" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color yellow', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="yellow" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color red', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="red" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render with color void', () => {
  const component = renderer.create(
    <ColourThemeSwatch color="void" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
