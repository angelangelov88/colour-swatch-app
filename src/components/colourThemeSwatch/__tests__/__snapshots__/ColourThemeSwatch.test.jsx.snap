// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with color blue 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-blue-60 text-blue-80 active:text-blue-90 border-blue-80 outline-blue-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color cyan 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-cyan-60 text-cyan-80 active:text-cyan-90 border-cyan-80 outline-cyan-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color gray 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-gray-60 text-gray-80 active:text-gray-90 border-gray-80 outline-gray-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color green 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-green-60 text-green-80 active:text-green-90 border-green-80 outline-green-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color lightblue 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-lightblue-60 text-lightblue-80 active:text-lightblue-90 border-lightblue-80 outline-lightblue-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color primary 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-primary-60 text-primary-80 active:text-primary-90 border-primary-80 outline-primary-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color purple 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-purple-60 text-purple-80 active:text-purple-90 border-purple-80 outline-purple-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color red 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-red-60 text-red-80 active:text-red-90 border-red-80 outline-red-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color teal 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-teal-60 text-teal-80 active:text-teal-90 border-teal-80 outline-teal-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color void 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-white text-gray-60  border-gray-60 outline-primary-30 active:border-gray-80 active:text-gray-90 hover:text-gray-70"
  onClick={[Function]}
  type="button"
>
  <span
    className="w-5 h-5"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M18.364 18.364A9 9 0 005.636 5.636m12.728 12.728A9 9 0 015.636 5.636m12.728 12.728L5.636 5.636"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;

exports[`should render with color yellow 1`] = `
<button
  className="flex items-center justify-center w-6 h-6 rounded border group hover:cursor-pointer hover:outline hover:outline-4 active:shadow-[inset_0_0_4px_rgba(0,0,0,0.5)] hover:shadow-[0_1px_3px_rgba(112,53,13,0.25)] bg-yellow-60 text-yellow-80 active:text-yellow-90 border-yellow-80 outline-yellow-30"
  onClick={[Function]}
  type="button"
>
  <span
    className="group-hover:block w-5 h-5 hidden"
  >
    <svg
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      strokeWidth={2}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </span>
</button>
`;
